Modifications to polar graphs

Carifio24 opened the pull request:
This PR implements the changes to polar scatterplot styling discussed at the glue meeting on 1/13/2022. In particular, these changes are:

Rather than display axis labels along the bottom and side of the plot (which doesn't really make sense for a polar plot), the axis labels are included in the angular and radial tick marks. In particular, the angular tick at zero angle is of the form `label=0`, while the outermost radial tick has the form `label=<value>`. Additionally, the radial tick marks are italicized to make the two sets of ticks clearly distinct.
![polar_plot](https://user-images.githubusercontent.com/14281631/152288871-ee0d5906-7015-4087-9515-e3835c58eb72.png)

If the label is an empty string, the equals is omitted:
![polar_plot_empty_label](https://user-images.githubusercontent.com/14281631/152288892-c64e98cf-2b80-415d-8bea-0ee560247379.png)

When in polar mode, the axis widget is adjusted accordingly:
![axes_options](https://user-images.githubusercontent.com/14281631/152288905-a7c374e8-96a8-40f0-aceb-87f2c709390e.png)

astrofrog commented on glue/viewers/scatter/viewer.py:
```diff
@@ -26,39 +26,52 @@ def setup_callbacks(self):
-            elif self.using_polar():
-                self.state.x_axislabel = ""
+            # elif self.using_polar():
+            #     self.state.x_axislabel = ""
```
Can this code now be removed?

Carifio24 replied:
Yes, has been removed

astrofrog commented on glue/viewers/scatter/viewer.py:
```diff
@@ -26,39 +26,52 @@ def setup_callbacks(self):
-            elif self.using_polar():
-                self.state.y_axislabel = ""
+            # elif self.using_polar():
+            #     self.state.y_axislabel = ""
```
Can this be removed?

Carifio24 replied:
Yes, has been removed

astrofrog reviewed (changes requested):
Looks good! Just a couple of small comments - also the ``codestyle`` CI job is failing and that is real, so could you fix the errors? (``tox -e codestyle`` to fix locally)

astrofrog commented:
Could you also rebase this to pick up the latest CI fixes?

Carifio24 commented:
I've rebased these commits onto the current master and added a changelog entry. Looks like all of the CI is passing except codecov/project. I'm not sure what the best way to handle that is.

astrofrog commented:
@Carifio24 - I think we can ignore the coverage failure, most changes here are covered (as shown by the codecov/patch job)
